File: AltCover.Recorder/InstrumentationAttribute.fs
﻿namespace AltCover.Recorder

open System
open System.Diagnostics.CodeAnalysis

/// <summary>
/// An attribute to label an instrumented assembly by provenance
/// </summary>
[<AttributeUsage(AttributeTargets.Assembly)>]
[<Sealed>]
type InstrumentationAttribute() =
  class
    inherit Attribute()

    /// <summary>
    /// SHA-256 hash of the original assembly
    /// </summary>
    member val Assembly = "AltCover.Recorder.g!" with get, set

    /// <summary>
    /// SHA-256 hash of instrumentation parameters
    /// </summary>
    member val Configuration = "Uninstrumented!!" with get, set
  end

File: AltCover.Recorder/InstrumentationAttribute.cs
namespace AltCover.Recorder
{
  using System;

  /// <summary>
  /// An attribute to label an instrumented assembly by provenance
  /// </summary>
  [AttributeUsage(AttributeTargets.Assembly)]
  public sealed class InstrumentationAttribute : Attribute
  {
    /// <summary>
    /// SHA-256 hash of the original assembly
    /// </summary>
    public string Assembly { get; set; }

    /// <summary>
    /// SHA-256 hash of instrumentation parameters
    /// </summary>
    public string Configuration { get; set; }

    public InstrumentationAttribute()
    {
      Assembly = "AltCover.Recorder.g!";
      Configuration = "Uninstrumented!!";
    }
  }
}